### PR TITLE
Make responsive status bar

### PR DESCRIPTION
**Pull request by jevenski**
Made a responsive status bar according to previous discussion in this [PR](https://github.com/Novik/ruTorrent/pull/2687).
The CPU and disk usage meters and the port status indicator haven't been curated yet, and will be in the comming commits. Below are some screenshots:

1. Mobile screens (smaller than 768px):

With more potential status bar cells added by plugins, showing infinite growing lines upwards (simulated now as a test)

2. Middle-sized screens (between 768px and 992px):
Some unnecessary information is hidden like speed limit, total traffic, system time, etc, but most cells are still displayed.

4. Large screens (992px and up):
Showing everything that's available, pushing system time to the right end.

### CODE REVIEW

**Comment by stickz** [comment]
**We have more room on the bottom status bar than I thought.**
1. Let's reduce the size of the CPU meter by 50%.
2. Let's remove the word "Speed:" for upload and download.
3. Let's fit everything on the bottom bar expect for port checker.
4. Let's disable port checker on mobile display.

**Comment by jevenski** [comment]
>     3. Let's fit everything on the bottom bar expect for port checker.
> 
>     4. Let's disable port checker on mobile display.

So, just want to sure, we leave CPU meter and disk meter on, and place them on the same single line alongside the other status cells, and turn port checker off on mobile display. And we keep everything as they are on desktop display. Is that correct?

**Comment by stickz** [comment]
> > ```
> > 3. Let's fit everything on the bottom bar expect for port checker.
> > 
> > 4. Let's disable port checker on mobile display.
> > ```
> 
> So, just want to sure, we leave CPU meter and disk meter on, and place them on the same single line alongside the other status cells, and turn port checker off on mobile display. And we keep everything as they are on desktop display. Is that correct?

Yes.

**Comment by jevenski** [comment]
All the status cells are now placed in a single line now. The three status bar related plugins were adjusted as well. Screenshots are shown below. Note that the disk usage meter is set to 100px long for desktop screens and 50px long for smaller screens, but the CPU load meter is always 50px wide. That's because the CPU load meter is an HTML `canvas` generated by JQuery flot, and is more complicated to adjust the size correctly when switching between different resolutions.

1. Desktop screens (wider than 992px):


2. Middle-sized screens (between 768px and 992px):


3. Mobile screens (narrower than 768px, below is 375px wide simulated by Chrome):

**Comment by stickz** [comment]
I think we should create 2 CPU graphs. One for desktop and one for mobile. Then set the display attribute inside a media query. We can still have 1 AJAX call and add the data to both the graphs on success. This will not incur very much overhead.

```js
plugin.check = function () {
  $.ajax({
    type: "GET",
    timeout: theWebUI.settings["webui.reqtimeout"],
    async: true,
    cache: false,
    url: "plugins/cpuload/action.php",
    dataType: "json",
    cache: false,
    success: function (data) {
      plugin.graphDesktop.addData(data.load);
      plugin.graphMobile.addData(data.load);
    },
  });
};

plugin.init = function () {
  if (getCSSRule("#meter-cpu-holder")) {
    plugin.prgStartColor = new RGBackground("#99D699");
    plugin.prgEndColor = new RGBackground("#E69999");
    plugin.addPaneToStatusbar(
      "meter-cpu-pane-desktop",
      $("<div>").append(
				$("<div>").addClass("icon"),
				$("<div>").attr({id: "meter-cpu-holder-desktop"}),
			),
			0, true,
    );
    plugin.addPaneToStatusbar(
      "meter-cpu-pane-mobile",
      $("<div>").append(
				$("<div>").addClass("icon"),
				$("<div>").attr({id: "meter-cpu-holder-mobile"}),
			),
			0, true,
    );
    plugin.graphDesktop = new rLoadGraph();
    plugin.graphDesktop.create($("#meter-cpu-holder-desktop"));
    plugin.graphMobile = new rLoadGraph();
    plugin.graphMobile.create($("#meter-cpu-holder-mobile"));
    plugin.check();
    plugin.reqId = theRequestManager.addRequest("ttl", null, plugin.check);
    plugin.markLoaded();
  } else window.setTimeout(arguments.callee, 500);
};
```
```css
#meter-cpu-pane .icon {
  background: url(./images/cpu.png) center no-repeat;
}
#meter-cpu-holder-desktop {
  position: relative;
  height: 16px;
  width: 100px;
}
#meter-cpu-holder-mobile {
  position: relative;
  height: 16px;
  width: 50px;
}
#meter-cpu-text {
  position: absolute;
  width: 100%;
  height: 100%;
  text-align: center;
}

.meter-cpu-pane-desktop { display: none; }
.meter-cpu-pane-mobile { display: block; }

/* Small devices (landscape phones, 576px and up) */
@media (min-width: 576px) { 
  .meter-cpu-pane-desktop { display: block; }
  .meter-cpu-pane-mobile { display: none; }
}
```

**Comment by jevenski** [comment]
I found a solution. Bind a resize handler to the window resize event and everything goes smoothly.

**Comment by stickz** [comment]
Sorry for the delay. I was fixing a bunch of issues with rTorrent.

The cpu meter is fixed. However, there is an issue with the port checker not displaying the ip address and port. I would also like to know if we can add the green check mark or red x on the mobile display. Then omit the ip address and port for mobile.

**Comment by jevenski** [comment]
No worries and take yor time. I thought you were out on vacation, maybe went to the Euro Cup final😀 Just kidding.

I did a small tweak with the port checker, making the IP address censored by default. Clicking on the IP address can toggle the display. That's mostly because I had to edit to blur out the IP address when posting a screenshot, and I sometimes forgot to do so. If that's a feature unwanted, I can have it removed.

**Comment by stickz** [comment]
> I did a small tweak with the port checker, making the IP address censored by default. Clicking on the IP address can toggle the display. That's mostly because I had to edit to blur out the IP address when posting a screenshot, and I sometimes forgot to do so. If that's a feature unwanted, I can have it removed.
> 

Yeah, I think we should remove this feature.

File: plugins/check_port/init.js [hunk 1]
```diff
@@ -46,41 +45,20 @@ plugin.createPortMenu = function(e)
...
 {
 	plugin.addPaneToStatusbar(
 		"port-pane",
 		$("<div>").append(
 			$("<div>").addClass("icon"),
-			$("<span>").attr("id","port-ip-text"),
+			$("<span>").attr("id","port-ip-text").addClass("d-none d-lg-block"),
 		),
-		-1, false,
+		-1, true,
```
Comment (jevenski) [inline plugins/check_port/init.js:56]: Port status indicator icon is mobile visible by setting `true`.

File: plugins/check_port/init.js [hunk 2]
```diff
@@ -46,41 +45,20 @@ plugin.createPortMenu = function(e)
...
-}
-
 plugin.onLangLoaded = function()
 {
 	plugin.addPaneToStatusbar(
 		"port-pane",
 		$("<div>").append(
 			$("<div>").addClass("icon"),
-			$("<span>").attr("id","port-ip-text"),
+			$("<span>").attr("id","port-ip-text").addClass("d-none d-lg-block"),
```
Comment (jevenski) [inline plugins/check_port/init.js:54]: IP address is hidden on small screens and begin to show up on large screens (>=992px wide).

**Comment by stickz** [comment]
Looks good, thanks!

**Pull request merged**